base/math.jl

```diff
@@ -146,6 +146,10 @@ sqrt(x::Float32) = box(Float32,sqrt_llvm(unbox(Float32,x)))
 sqrt(x::Real) = sqrt(float(x))
 @vectorize_1arg Number sqrt
 
+# 1-arg hypot
+hypot(x::Real) = abs(promote(float(x))...)
```

Comment:
> 160: in this case why not just add the condition isnan(y) and just return NaN , instead of forcing it to go through the more expensive procedure .

Do you mean something like
```julia
hypot(x::Real) = isnan(x) ? x : abs(promote(float(x))...)
```
?  I'm not sure this is great improvement, but I'm no expert

> Also you could use abs2 here as well for consistency with the three argument version.

Why I should use `abs2` if the distance from 0 in 1D is just the absolute value (not squared) of the number?